ios/brave-ios/Sources/Brave/Frontend/Browser/HomePanel/NTPDataSource.swift

```diff
@@ -104,114 +110,116 @@ public class NTPDataSource {
   // This can 'easily' be adjusted to support both sets by switching to String, and using filePath to identify uniqueness.
   private var lastBackgroundChoices = [Int]()
 
-  private enum ImageRotationStrategy {
-    /// Special strategy for sponsored images, uses in-memory property to keep track of which image to show.
-    case sponsoredRotation
-    /// Uses random images, keeps in-memory list of recently viewed images to avoid showing it too often.
-    case randomOrderAvoidDuplicates
+  func shouldAttemptSponsoredMedia() -> Bool {
+    return
+      Preferences.NewTabPage.backgroundMediaType.isSponsored
+      && Preferences.NewTabPage.backgroundRotationCounter.value
+        == service.initialCountToBrandedWallpaper
   }
 
-  func newBackground() -> NTPWallpaper? {
-    if !Preferences.NewTabPage.backgroundImages.value { return nil }
+  func getSponsoredMediaBackground() -> NTPWallpaper? {
+    guard let sponsoredImageData = service.sponsoredImageData,
+      let newTabPageAd = rewards?.ads.maybeGetPrefetchedNewTabPageAd()
+    else { return nil }
+
+    let isSponsoredVideoAllowed =
+      Preferences.NewTabPage.backgroundMediaType == .sponsoredImagesAndVideos
+
+    for campaign in sponsoredImageData.campaigns {
+      if campaign.campaignId != newTabPageAd.campaignID {
+        continue
+      }
+
+      for creative in campaign.backgrounds {
+        if creative.logo.imagePath != nil
+          && creative.creativeInstanceId == newTabPageAd.creativeInstanceID
+          && (!creative.isVideoFile || isSponsoredVideoAllowed)
+        {
+          return NTPWallpaper.sponsoredMedia(creative)
+        }
+      }
+    }
+
+    rewards?.ads.onFailed(
+      toPrefetchNewTabPageAd: newTabPageAd.placementID,
+      creativeInstanceId: newTabPageAd.creativeInstanceID
+    )
+    return nil
+  }
 
+  func getImageBackground() -> NTPWallpaper? {
     // Identifying the background array to use
-    let (backgroundSet, strategy) = {
-      () -> ([NTPWallpaper], ImageRotationStrategy) in
+    let backgroundSet = {
+      () -> [NTPWallpaper] in
 
       if let theme = service.superReferralImageData,
         case let refCode = service.superReferralCode,
         !refCode.isEmpty,
         Preferences.NewTabPage.selectedCustomTheme.value != nil
       {
-        return (
+        return
           theme.campaigns.flatMap(\.backgrounds).map {
             NTPWallpaper.superReferral($0, code: refCode)
-          }, .randomOrderAvoidDuplicates
-        )
-      }
-
-      if let sponsor = service.sponsoredImageData {
-        let attemptSponsored =
-          Preferences.NewTabPage.backgroundMediaType.isSponsored
-          && Preferences.NewTabPage.backgroundRotationCounter.value
-            == service.initialCountToBrandedWallpaper
-          && !privateBrowsingManager.isPrivateBrowsing
```

Comment:
private tabs getting NTP SI now? The AdsService isn't ever going to be created with an OTR profile if thats what we're relying on now

Reply:
Added this check back:
https://github.com/brave/brave-core/pull/28233/files#diff-f177508d7aa697324ba2c4c67f5a208562b11f09191da96c8e64eff9494e9002R118